.github/workflows/release-porting-guide.yml

```
@@ -0,0 +1,54 @@
+---
+name: Ansible porting guide
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-version:
+        description: Release version. For example, 11.1.0
+        required: true
+      ansible-major-version:
+        description: Major version. For example, 11
+        required: true
+
+jobs:
+  upload-porting-guide:
+    name: Extract the porting guide${{ '' }}  # Nest jobs under the same sidebar category
+    runs-on: ubuntu-latest
+    env:
+      package-url: https://files.pythonhosted.org/packages/source/a/ansible
+      ansible-tarball: ansible-${{ inputs.ansible-version }}.tar.gz
+      extracted-path: ansible-${{ inputs.ansible-version }}
+      porting-guide-rst: porting_guide_${{ inputs.ansible-major-version }}.rst
```

Environment variables should always be written upper case and use `_` instead of `-`.
```suggestion
      package-url: https://files.pythonhosted.org/packages/source/a/ansible
      ansible-tarball: ansible-${{ inputs.ansible-version }}.tar.gz
      extracted-path: ansible-${{ inputs.ansible-version }}
      porting-guide-rst: porting_guide_${{ inputs.ansible-major-version }}.rst
```
```suggestion
      PACKAGE_URL: https://files.pythonhosted.org/packages/source/a/ansible
      ANSIBLE_LATEST: ansible-${{ inputs.ansible-version }}.tar.gz
      EXTRACTED_PATH: ansible-${{ inputs.ansible-version }}
      PORTING_GUIDE_RST: porting_guide_${{ inputs.ansible-major-version }}.rst
```